File: quarkus-weather-app/src/main/java/functions/Methods.java
package functions;

public class Methods
{
    public double kelvin_to_celsius(double temp)
    {
        double temp_celsius = 0.0d;
        temp_celsius = temp - 273.15;
        return temp_celsius;
    }

    public double kelvin_to_fahrenheit(double temp)
    {
        double temp_fahrenheit = 0.0d;
        temp_fahrenheit = (temp * 1.8) - 459.67;
        return temp_fahrenheit;
    }
}

File: functions-introduction/solutions/weather-python/methods.py
from pathlib import Path
import json

def read_weather(city: str):
    weather = {}
    filepath = Path(__file__).parent.joinpath("cities.json")

    try:
        with open(filepath) as data_file:
            weather = json.load(data_file)

    except FileNotFoundError as e:
        print("No cities.json file exists!!")
        raise e

    city = weather['city'].get(city.lower())

    return city



def kelvin_to_celsius(temperature: str):
    temp_celsius = (temperature - 273.15)
    temp_celsius = float(temp_celsius)

    return temp_celsius


def kelvin_to_farhenheit(temperature: str):
    temp_fahrenheit = (temperature * 1.8) - 459.67
    temp_fahrenheit = float(temp_fahrenheit)

    return temp_fahrenheit
